Source/WebCore/loader/cache/MemoryCache.cpp

```diff
@@ -452,9 +457,6 @@ void MemoryCache::remove(CachedResource& resource)
         }
     }
     RELEASE_ASSERT(!resource.inCache());
-
-    if (resource.canDelete())
-        resource.deleteThis();
```

Comment:
Why is it okay to stop deleting the resource here? I guess because we are protecting it at the top of the method?

We assert that the resource is no longer in the cache, and when we depart the method it should release the protection. But if we used to need to delete the resource, when does that happen now?

It seems like we should release the protection, then continue the delete like before, shouldn't we?